zephyr/lib/cpu.c

```diff
@@ -143,12 +143,19 @@ int cpu_enable_core(int id)
 		return 0;
 
 #if ZEPHYR_VERSION(3, 0, 99) <= ZEPHYR_VERSION_CODE
-	/* During kernel initialization, the next pm state is set to ACTIVE. By checking this
+	/* Zephyr implementations on secondary cores initialzation are
+	 * different between MTL+ and other platforms.
+	 *
+	 * On MTL+ platforms, only run z_init_cpu() on boot, not when waking up from D3,
+	 * regardless whether saving to IMR is enabled or not.
```

Comment:
that seems wrong to me @mengdonglin 

What is the difference between 'boot'  and 'waking up from D3' if there is no context saved in IMR?